.github/workflows/services-base.yml

```diff
@@ -0,0 +1,123 @@
+on:
+  push:
+    branches:
+      - 'main'
+    paths:
+      - .github/workflows/base-image.yml
+      - docker/base.Dockerfile
+  pull_request:
+    paths:
+      - .github/workflows/base-image.yml
+      - docker/base.Dockerfile
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: blockscout/services-base
+
+jobs:
+  check_tag:
+    name: Check tag existence
+    runs-on: ubuntu-latest
+    outputs:
+      tag: ${{ steps.get_image_tag.outputs.tag }}
+      is_new: ${{ steps.check_existence.outputs.is_new }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Get image tag
+        id: get_image_tag
+        env:
+          DOCKERFILE_PATH: "docker/base.Dockerfile"
+        run: |
+          CONTENT=$(cat $DOCKERFILE_PATH)
+    
+          nl=$'\n'
+    
+          if [[ $CONTENT =~ ARG\ CARGO_CHEF_VERSION=([^$nl]+) ]]; then
+            CARGO_CHEF_TAG=${BASH_REMATCH[1]}
+          else
+            echo "Failed to extract CARGO_CHEF_TAG from Dockerfile"
+            exit 1
+          fi
```

Comment:
Do you think it's better to define versions in env and pass them down to docker using build args, instead of parsing a Dockerfile like this?

Reply:
I'm not sure, tbh. I like the current approach because all arguments are explicit and located in the same place as their usage. But I'm not insisting on that